### DISCUSSION TITLE
🐛 Side-by-side looking ugly on GitHub Actions log

### DISCUSSION THREAD

**Discussion by jpfeuffer** [dandavison/delta#1313] [General]
![image](https://user-images.githubusercontent.com/8102638/220414943-e94868e2-0b8f-41b7-90fb-d41999279a61.png)

see #1309 for the diff
https://github.com/OpenMS/pyopenms-docs/actions/runs/4234240051/jobs/7356257942#step:7:152

Am I missing a setting?

**Comment by dandavison** [comment]
If I save your diff from #1309 and then do

```
cat /tmp/z.diff | delta --no-gitconfig --side-by-side
```

it looks normal. It looks like you might be trying to use `bat` as delta's pager? Check the output of
```
env | grep PAGER
```

delta's pager should be `less` not `bat`.

**Comment by jpfeuffer** [comment]
So, I double-checked. The PAGER variable was not set at all.
But even if I set it to `less` it does not work.

It works on my local computer, too. But not on the GitHub Actions logs.

**Comment by dandavison** [comment]
Hm, my guess is the automatic terminal width detection. _If_ you felt like more investigations (!) you could look into things like `echo $COLUMNS` and `tput cols` and see if they are something sane.

**Comment by jpfeuffer** [comment]
By default, the environment of a GitHub Actions step is almost completely empty. This means also COLUMNS is missing.
Also after setting COLUMNS to 150, trying with and without "--width 150", with --line-fill-method ansi|spaces.
Nothing works.
As soon as there is a plus or minus line in the git output, your tool just adds _no padding at all_ between left and right panel.

Tput output:
`tput: No value for $TERM and no -T specified`

**Comment by jpfeuffer** [comment]
If you check the screenshot, you will see that it works for lines with no changes and lines that were completely added/removed.
So I am suspecting a bug in delta.

**Comment by jpfeuffer** [comment]
> Hm, my guess is the automatic terminal width detection

Why would it need to do that in the fixed "--width 150" case?
EDIT: I guess for wrapping, hmm..

**Comment by jpfeuffer** [comment]
I just realized not only side-by-side is affected!

![image](https://user-images.githubusercontent.com/8102638/220450325-cbb0b8ff-b171-4678-a9df-b5145f7178cc.png)

It does not print *any* leading whitespaces in added lines.

Note: I had to switch to `git diff --diff-algorithm histogram | delta` because of the problems in #1309

**Comment by jpfeuffer** [comment]
Setting:
```
env:
    COLUMNS: 150
    LINES: 10000
    TERM: xterm-256color
```
as expected, does not help either. It shows correctly with `tput cols` now, as 150.

**Comment by dandavison** [comment]
Hm, maybe a bug in ANSI sequence handling by the terminal emulator being used by the Github Actions log UI? If you can't reproduce this in a standard terminal emulator then it seems like there's no evidence of a delta bug here, right?
